evm_loader/lib/src/commands/emulate.rs

```diff
@@ -182,6 +183,8 @@ async fn initialize_storage_and_transaction<'rpc, T: Rpc + BuildConfigSimulator>
     program_id: &Pubkey,
     emulate_request: &EmulateRequest,
     rpc: &'rpc T,
+    do_increment_nonce: bool,
+    do_transfer_gas_limit: bool,
```

Comment:
I don't like the `flag` arguments which change the function internal logic. It's hard to keep in mind all possible combinations. I think in this case, the code will be clearer if you simply call the `increment_nonce` and `transfer_gas_limit` functions in the initialization places.

Reply:
fixed